App/components/Input/input.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useCallback, useRef } from 'react'
+import { TextInput } from 'react-native'
+import {
+  Container,
+  PlaceholderLabel,
+  LabeledInput,
+  RubleIcon,
+} from './input.styled'
+import Ruble from '../../../assets/images/ruble.svg'
+
+interface InputProps {
+  placeholder: string
+}
+
+interface InputReference extends TextInput {
+  value: string
+}
+
+const Input: React.FC<InputProps> = ({ placeholder, ...rest }) => {
+  const [isFocused, setIsFocused] = useState(false)
+  const [isFilled, setIsFilled] = useState(false)
+
+  const inputRef = useRef<InputReference>(null)
+
+  const handleInputFocus = useCallback(() => {
+    setIsFocused(true)
+  }, [])
```

Comment:
We should pass ```setIsFocused``` and such as a dependency inside useCallback

Reply:
done